app/src/main/java/com/hewgill/android/nzsldict/SignVideoFragment.java

```diff
@@ -83,9 +127,13 @@ public void onPrepared(MediaPlayer mp) {
             }
         });
 
+        // Start loading video if network is available
+        if (checkConnectivity()) mVideo.setVideoURI(Uri.parse(mDictItem.video));
```

Comment:
the name of this method `checkConnectivity` doesn't warn that it has side effects. 
It does more than check, it also changes the UI.